internal/scheduler/scheduler_metrics.go

```diff
@@ -0,0 +1,161 @@
+package scheduler
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/armadaproject/armada/internal/scheduler/interfaces"
+)
+
+const (
+	NAMESPACE = "armada"
+	SUBSYSTEM = "scheduler"
+)
+
+type SchedulerMetrics struct {
+	// Cycle time when scheduling, as leader.
+	scheduleCycleTime prometheus.Histogram
+	// Cycle time when reconciling, as leader or follower.
+	reconcileCycleTime prometheus.Histogram
+	// Number of jobs scheduled per queue.
+	scheduledJobs prometheus.HistogramVec
+	// Number of jobs preempted per queue.
+	preemptedJobs prometheus.HistogramVec
+}
+
+var (
+	schedulerMetricsPointer *SchedulerMetrics
+	singletonLock           sync.Mutex
+)
+
+func GetSchedulerMetrics() *SchedulerMetrics {
+	singletonLock.Lock()
+	defer singletonLock.Unlock()
+
+	if schedulerMetricsPointer == nil {
+		schedulerMetricsPointer = newSchedulerMetrics()
+	}
+
+	return schedulerMetricsPointer
+}
+
+func newSchedulerMetrics() *SchedulerMetrics {
+	scheduleCycleTime := prometheus.NewHistogram(
+		prometheus.HistogramOpts{
+			Namespace: NAMESPACE,
+			Subsystem: SUBSYSTEM,
+			Name:      "schedule_cycle_times",
+			Help:      "Cycle time when in a scheduling round.",
+			Buckets:   prometheus.LinearBuckets(0, 5, 20),
+		},
+	)
+
+	reconcileCycleTime := prometheus.NewHistogram(
+		prometheus.HistogramOpts{
+			Namespace: NAMESPACE,
+			Subsystem: SUBSYSTEM,
+			Name:      "reconcile_cycle_times",
+			Help:      "Cycle time when outside of a scheduling round.",
```

Comment:
This is the cycle time when not leader, no?

Reply:
When not leader or when not scheduling. That is when you're just reading the jobs to get up to date in the state.

Reply:
I'd say that in the description.